src/tools/launcher/java_launcher.cc

```diff
@@ -152,6 +152,9 @@ static void WriteJarClasspath(const wstring& jar_path,
       if (x == L' ') {
         *manifest_classpath << L"%20";
       }
+      if (x == L'%') {
```

Comment:
1. This code seem to skip jar_paths that have '%' in them and no spaces or back-slash. (because of first if) Is this intended?
2. If '%' occurs in the jar_path, then '%25%' seems to be the result in manifest_classpath. (because there is no else between ifs). Is this intended behaviour?

Reply:
You're right, the ordering of this is wrong. I'll need to change this here and elsewhere.

Reply:
@justhecuke-zz , do I understand correctly that this PR is still waiting on the resolution of this item?